pyogrio/tests/test_geopandas_io.py

```diff
@@ -207,6 +219,165 @@ def test_read_fids_force_2d(test_fgdb_vsi):
+        columns=["iso_a3_renamed", "name"],
+        where="iso_a3_renamed IN ('CAN', 'USA', 'MEX')",
+    )
+    assert len(df.columns) == 3
+    assert len(df) == 3
+    assert df.iso_a3_renamed.tolist() == ["CAN", "USA", "MEX"]
+
+
+def test_read_sql_columns_where_bbox(naturalearth_lowres_all_ext):
+    sql = "SELECT iso_a3 AS iso_a3_renamed, name, pop_est FROM naturalearth_lowres"
+    df = read_dataframe(
+        naturalearth_lowres_all_ext,
+        sql=sql,
+        sql_dialect="OGRSQL",
+        columns=["iso_a3_renamed", "name"],
+        where="iso_a3_renamed IN ('CAN', 'USA', 'MEX')",
+        bbox=(-140, 20, -100, 40)
+    )
+    assert len(df.columns) == 3
+    assert len(df) == 2
+    assert df.iso_a3_renamed.tolist() == ["USA", "MEX"]
+
+
+def test_read_sql_skip_max(naturalearth_lowres_all_ext):
+    sql = """SELECT *
+               FROM naturalearth_lowres
+              WHERE iso_a3 IN ('CAN', 'MEX', 'USA')
+              ORDER BY name"""
+    df = read_dataframe(
+            naturalearth_lowres_all_ext, sql=sql, skip_features=1,
+            max_features=1, sql_dialect="OGRSQL")
+    assert len(df.columns) == 6
+    assert len(df) == 1
+    assert df.iso_a3.tolist() == ["MEX"]
+
+    sql = "SELECT * FROM naturalearth_lowres LIMIT 1"
+    df = read_dataframe(
+            naturalearth_lowres_all_ext, sql=sql, max_features=3, sql_dialect="OGRSQL")
+    assert len(df) == 1
+
+    sql = "SELECT * FROM naturalearth_lowres LIMIT 1"
+    with pytest.raises(ValueError, match="'skip_features' must be between 0 and 0"):
+        _ = read_dataframe(
+                naturalearth_lowres_all_ext, sql=sql, skip_features=1, 
+                sql_dialect="OGRSQL")
+
+
+@pytest.mark.parametrize(
+        "naturalearth_lowres", [ext for ext in ALL_EXTS if ext != ".gpkg"],
+        indirect=["naturalearth_lowres"])
+def test_read_sql_dialect_sqlite_nogpkg(naturalearth_lowres):
+    # Should return singular item
+    sql = "SELECT * FROM naturalearth_lowres WHERE iso_a3 = 'CAN'"
+    df = read_dataframe(naturalearth_lowres, sql=sql, sql_dialect="SQLITE")
+    assert len(df) == 1
+    assert len(df.columns) == 6
+    assert df.iloc[0].iso_a3 == "CAN"
+    area_canada = df.iloc[0].geometry.area
+
+    # Use spatialite function
+    sql = """SELECT ST_Buffer(geometry, 5) AS geometry, name, pop_est, iso_a3
+               FROM naturalearth_lowres
+              WHERE ISO_A3 = 'CAN'"""
+    df = read_dataframe(naturalearth_lowres, sql=sql, sql_dialect="SQLITE")
+    assert len(df) == 1
+    assert len(df.columns) == 4
+    assert df.iloc[0].geometry.area > area_canada
+
+
+@pytest.mark.parametrize(
+        "naturalearth_lowres", [".gpkg"],
+        indirect=["naturalearth_lowres"])
+def test_read_sql_dialect_sqlite_gpkg(naturalearth_lowres):
+    # Should return singular item
+    sql = "SELECT * FROM naturalearth_lowres WHERE iso_a3 = 'CAN'"
+    df = read_dataframe(naturalearth_lowres, sql=sql, sql_dialect="INDIRECT_SQLITE")
```

Comment:
Suggest adding a comment indicating that `INDIRECT_SQLITE` uses spatialite functionality within GDAL rather than depending on libspatialite.